From 7d4444f71c30c9c434b5008c544db689bf9d9e13 Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Tue, 25 Dec 2018 09:10:30 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399: don't enable edp_hpd pin by
 default on soc dtsi

Not all the rk3399 boards use gpio4_c7 pin for edp_hpd.
So this pin should be specified in the board dts.
Without this patch we see below pin conflict when hdmi-cec and edp enabled.

[    0.969331] rockchip-pinctrl pinctrl: pin gpio4-23 already requested by
ff940000.hdmi; cannot claim for ff970000.edp
[    0.969350] rockchip-pinctrl pinctrl: pin-151 (ff970000.edp) status -22
[    0.969361] rockchip-pinctrl pinctrl: could not request pin 151 (gpio4-23)
from group edp-hpd on device rockchip-pinctrl

Change-Id: I324f28ea9f995ee4b84869ac369ab63e0c141cfa
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 .../boot/dts/rockchip/rk3399-evb-rev3-android-mipi-edp.dts     | 2 ++
 arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts        | 2 ++
 arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts        | 3 +++
 .../boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts      | 1 +
 arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts               | 2 ++
 arch/arm64/boot/dts/rockchip/rk3399.dtsi                       | 2 --
 6 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-mipi-edp.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-mipi-edp.dts
index a3094b54f63a..8ccfa594bfe8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-mipi-edp.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-mipi-edp.dts
@@ -228,6 +228,8 @@
 
 &edp {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&edp_hpd>;
 
 	ports {
 		edp_out: port@1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
index a8294924e17b..18bc090e8fa8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
@@ -458,6 +458,8 @@
 
 &edp {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&edp_hpd>;
 
 	ports {
 		edp_out: port@1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
index a63cd8a64479..139b1b31fa81 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
@@ -1046,6 +1046,9 @@
 
 &edp {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&edp_hpd>;
+
 	ports {
 		edp_out: port@1 {
 			reg = <1>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
index 4ce75841eb8a..82f34590ce03 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
@@ -280,6 +280,7 @@
 
 &edp {
 	status = "okay";
+	force-hpd;
 
 	ports {
 		edp_out: port@1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
index 21e33fae2cf7..9239bc4f2aac 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
@@ -413,6 +413,8 @@
 
 &edp {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&edp_hpd>;
 
 	ports {
 		edp_out: port@1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index b0a7bd4b507f..c1a5ac163bd5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -2110,8 +2110,6 @@
 		reset-names = "dp";
 		rockchip,grf = <&grf>;
 		status = "disabled";
-		pinctrl-names = "default";
-		pinctrl-0 = <&edp_hpd>;
 
 		ports {
 			#address-cells = <1>;
-- 
2.35.3

